src/haddock/gear/tables.py

```diff
@@ -0,0 +1,561 @@
+
+    .. code:: python
+
+        import numpy as np
+        np.loadtxt('table.txt', dtype=str)
+
+    Finally, if you want to retrieve the same data dictionary used to
+    create the table, see :py:func:`read_table_to_data_dict`.
+
+    Parameters
+    ----------
+    data : dict
+        A dictionary where keys are the column names and values are the
+        column values for each row. ``data`` dictionary values can be in
+        the form of a list if there are multiple rows, or a single value
+        if there is only one row.
+
+    header : None or str, optional
+        The header text to write before the table. Usually this is a commented
+        text consisting of a small report or instructions. Defaults to
+        ``None``, no header is written.
+
+    column_spacing : str, optional
+        The spacing between columns. Defaults to four empty spaces.
+
+    fmt_args : arguments
+        Named arguments to pass to
+        :py:func:`haddock.gear.tables.format_value`.
+
+    Returns
+    -------
+    str
+        Table formatted string.
+
+    See Also
+    --------
+    :py:func:`haddock.gear.tables.read_table_to_data_dict`
+    :py:func:`haddock.gear.tables.convert_adjusted_width_to_tsv`
+    :py:func:`haddock.gear.tables.convert_adjusted_width_to_csv`
+    """
+    header = "" if header is None else header
+    col_names = list(data.keys())
+    formatted_col_names = [None] * len(col_names)
+
+    for i, col_name in enumerate(copy(col_names)):
+
+        longest_ = max(
+            len(value_to_str(value))
+            for value in transform_to_list(data[col_name])
+            )
+        longest = max(longest_, len(col_name))
+
+        formatted_col_names[i] = col_name.center(longest, " ")
+
+        data[col_name] = [
+            format_value(value, longest, " ", **fmt_args)
+            for value in transform_to_list(data[col_name])
+            ]
+
+    col_name_row = column_spacing.join(formatted_col_names)
+
+    # all columns should have the same lenght
+    num_rows = len(data[col_names[0]])
+    rows = [None] * num_rows
+    for i in range(num_rows):
+        rows[i] = []
+        for col_name in col_names:
+            try:
+                rows[i].append(data[col_name][i])
+            except IndexError as err:
+                emsg = (
+                    f"Column {col_name!r} as less items than the "
+                    "expected number of rows."
+                    )
+                raise TableFormatError(emsg) from err
+        rows[i] = column_spacing.join(rows[i])
+    table_data = os.linesep.join(rows)
+
+    # to situations to save tables with and header to avoid having extra
+    # empty lines.
+    if header:
+        table = os.linesep.join([
+            header,
+            col_name_row,
+            table_data,
+            os.linesep,
+            ])
+    else:
+        table = os.linesep.join([
+            col_name_row,
+            table_data,
+            os.linesep,
+            ])
```

Comment:
I think it's better to create the overall table first:
```
table = os.linesep.join([
            col_name_row,
            table_data,
            os.linesep,
            ])
```
and then check for the existence of the header: shorter code, better readability, do you agree?